### PR TITLE
chore: update docusaurus to 3.X

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/guides/cli/20_proxy-and-tunnel.mdx [hunk 1]
```diff
@@ -63,8 +63,18 @@ production, it's recommended to use a CNAME.
...
+```
 
+<Mermaid
+  chart={`
+flowchart TD
+    A[Developer] -->|GET http://localhost:1234| B(Ory Proxy)
+    B -->|GET /.ory/*| D[Ory Network]
+    B -->E[Your application]
+`}
+/>
```
Comment (aeneasr) [inline docs/guides/cli/20_proxy-and-tunnel.mdx:77]: Code fencing can help a lot with MDX problems:

`````
````mdx-code-block
 import Mermaid from "@site/src/theme/Mermaid"

 <Mermaid
   chart={`
 flowchart TD
     A[Developer] -->|GET http://localhost:1234| B(Ory Proxy)
     B -->|GET /.ory/*| D[Ory Network]
     B -->E[Your application]
 `}
 />
````
`````

**Comment by jonas-jonas** [comment]
Requires https://github.com/ory/ci/pull/148

File: .github/workflows/argos.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
     branches: [master]
   pull_request:
     branches: [master]
+  workflow_dispatch:
```
Comment (piotrmsc) [inline .github/workflows/argos.yml:8]: Do we need to run it on demand?

File: docs/kratos/social-signin/30_apple.mdx [hunk 3]
```diff
@@ -230,3 +244,7 @@ import SocialSigninTroubleshooting from '../_common/social-sign-in-troubleshooti
 
 Make sure to either add your apps current identifier to the `additional_id_token_audiences` field or set it as the Client ID of
 the provider in the Ory Console.
+
+```
```
Comment (piotrmsc) [inline docs/kratos/social-signin/30_apple.mdx:248]: probably can be removed?

File: docs/kratos/social-signin/30_apple.mdx [hunk 4]
```diff
@@ -16,22 +16,25 @@ one.
...
 
 Follow these steps to add Apple as a social sign-in provider to your project using the Ory Console:
 
 1. Sign in to Ory Console and select [**Social Sign-in**](https://console.ory.sh/projects/current/social-signin).
 2. Click the switch next to the Apple logo to start the configuration.
 3. Copy the Redirect URI and save it for later use.
 4. Using an Apple Developer Account, create an app, a service, and a private key:
-   1. To set up an application, navigate to the Apple Developer Dashboard and go to [Certs, Identifiers, and Profiles then Identifiers](https://developer.apple.com/account/resources/identifiers/list).
-   2. Create a new [**App IDs** identifier](https://developer.apple.com/account/resources/identifiers/add/bundleId). When prompted for a type select **App**.
+   1. To set up an application, navigate to the Apple Developer Dashboard and go to
```
Comment (piotrmsc) [inline docs/kratos/social-signin/30_apple.mdx:34]: is it possible now with docusaurus v3 to respect ordered lists? 
it renders like this. : 
Screenshot 2024-06-10 at 12 37 54

Comment (jonas-jonas) [reply docs/kratos/social-signin/30_apple.mdx:34]: > to respect ordered lists

What do you mean by this? It's a numbered ordered list, but the second level is using roman numerals, to make the distinction.

Comment (piotrmsc) [reply docs/kratos/social-signin/30_apple.mdx:34]: yeah, I was thinking of not using roman numerals

File: .github/workflows/argos.yml (hunk 2 above)
Comment (jonas-jonas) [reply .github/workflows/argos.yml:8]: It's disabled right now. I want to be able to dispatch it manually though

File: docs/kratos/social-signin/30_apple.mdx (hunk 4 above)
Comment (jonas-jonas) [reply docs/kratos/social-signin/30_apple.mdx:34]: I don't really see anything in regard to that. But we can probably break up this section to not have a nested list in there.

File: docs/kratos/social-signin/30_apple.mdx (hunk 3 above)
Comment (jonas-jonas) [reply docs/kratos/social-signin/30_apple.mdx:248]: yup, not sure where it came from...

File: docs/kratos/self-service/flows/user-registration.mdx [hunk 5]
```diff
@@ -18,7 +18,6 @@ import {
 } from "./code/registration"
 import RenderFlow from "@site/src/theme/Code/RenderFlow"
 import DomainWarning from "@site/src/theme/DomainWarning"
-import AjaxWarning from "@site/src/theme/AjaxWarning"
```
[the remark this answers is not in the document]
Comment (vinckr) [reply docs/kratos/self-service/flows/user-registration.mdx:21]: Lets create some followup issues after we merge this to keep track?

**Review by vinckr — APPROVED** [review]
lgtm :)
looked through most pages

**Comment by aeneasr** [comment]
I didn't look at the 60k lines changed but browsed a few pages and they look good to me. Did you check that there are no page speed or SEO issues?

**Comment by jonas-jonas** [comment]
@aeneasr most of that is package-lock.json:
 
image

Performance is a little worse on the preview env, but that might be due to it being a preview env, SEO is 100%.

**Comment by aeneasr** [comment]
Yeah, preview env is always a bit slower

**Review by aeneasr — APPROVED** [review]

**Review by vinckr — APPROVED** [review]

**Pull request merged**